js/modules/loki_snode_api.js

```diff
@@ -249,11 +264,85 @@ class LokiSnodeAPI {
     if (this.randomSnodePool.length === 0) {
       throw new window.textsecure.SeedNodeError('Invalid seed node response');
     }
+    // FIXME: _.sample?
     return this.randomSnodePool[
       Math.floor(Math.random() * this.randomSnodePool.length)
     ];
   }
 
+  // use nodes that support more than 1mb
+  async getRandomProxySnodeAddress() {
+    /* resolve random snode */
+    if (this.randomSnodePool.length === 0) {
+      // allow exceptions to pass through upwards
+      await this.refreshRandomPool();
+    }
+    if (this.randomSnodePool.length === 0) {
+      throw new window.textsecure.SeedNodeError('Invalid seed node response');
```

Comment:
Should we throw this in `refreshRandomPool()` instead?

Reply:
Not sure, @Mikunj do you know if this is an issue that we throw inside a promise instead of the outer function?

Reply:
Throwing inside `async` should reject the promise and `await`ing on rejected promise should just re-throw the original exception.

Reply:
which we catch on line 483 and rethrow